### ISSUE TITLE
Select the directory of where to generate the client?

### ISSUE THREAD

**Issue by birgersp** [prisma/prisma#13123]
## Problem

Consider two directories:

* shared
* project

My schema.prisma resides in the `shared` directory. There is also a `package.json` in the shared directory. I am trying to generate a client in the `project` directory from the schema.prisma file.

If i `cd` to `project` and do `npx prisma generate --schema=../shared/prisma.schema` then the client is built in `shared`, and not in `project` like I wanted.

## Suggested solution

Add a flag to specify working directory of the `generate` command. Use the working directory to generate the client, only use the `--schema` flag to read the file but ignore it when deciding where to put the client.

## Alternatives

(None)

## Additional context

![image](https://user-images.githubusercontent.com/5260237/166430264-8873c109-95ec-48e0-9ea1-dc15e8c2c4b6.png)

**Comment by Pilfer** [comment]
Couldn't you create a new repo/module that contains and exports your generated client, then import it into your project(s) that share the schema and client? It might suck when it comes to running migrations and pinning versions, but it might yield the results you're looking for.

**Comment by birgersp** [comment]
As a workaround? Probably yes.

Is that an argument against the requested feature?

**Comment by Pilfer** [comment]
> > Couldn't you create a new repo/module that contains and exports your generated client, then import it into your project(s) that share the schema and client? It might suck when it comes to running migrations and pinning versions, but it might yield the results you're looking for.
> 
> As a workaround? Probably yes. Is that an argument against the requested feature?

Just a workaround, of course! :)

I figured you might like to stay productive until the maintainers have the opportunity to look over your issue/feature request.

**Comment by janpio** [comment]
Is the `output` property maybe already what you are looking for?
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#fields-1

**Comment by birgersp** [comment]
> Just a workaround, of course! :)
> 
> I figured you might like to stay productive until the maintainers have the opportunity to look over your issue/feature request.

I did a similiar setup to this ([link](https://isidoro-ferreiro.medium.com/share-your-prisma-client-across-projects-44d1c7aca6fd)), but some of the projects that was supposed to share the "prisma client lib" didn't really work with referencing the lib so it ended up being easier to just keep a copy of `schema.prisma` in them. Anyways, thanks for the tip :)


> Is the `output` property maybe already what you are looking for? https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#fields-1

Thanks, but no. The idea here is that I want to keep a `schema.prisma` file that is shared across multiple projects. So in the context of `schema.prisma` it isn't known where the libs should be generated.

**Comment by janpio** [comment]
Interesting use case.

A workaround might be to have _multiple_ `generator` blocks with different `output` folder in your schema file. That would have the side effect of creating a new client for all of the output folders when you run `prisma generate` though - not sure if that is a problem or even welcome.

**Comment by birgersp** [comment]
> Interesting use case.
> 
> A workaround might be to have _multiple_ `generator` blocks with different `output` folder in your schema file. That would have the side effect of creating a new client for all of the output folders when you run `prisma generate` though - not sure if that is a problem or even welcome.

I will try this. My initial thought is that this might be a nice workaround, because it would ensure that all project that uses the shared schema are always "synced" when the schema is changed (and `prisma generate` is run).

**Comment by janpio** [comment]
Let us know if this works. If it does, we might want to document your use case so it is more discoverable that this is an option that makes sense.

**Comment by antoineanorra** [comment]
I also have this problem. I have a dashboard & a site that share the same prisma.schema file. They are in two separate folders, so to avoid redundancy of information, I put the prisma file in another folder so that it is common.

So, with your solution, my prisma file has 2 generators, 1 for dashboard & 1 for the site. 
It works but it creates a node_module & 2 package*.json in the db folder and not in dashboard/site. 

We agree that by doing this, I have to import prisma with the files that were created by the generator?

Is @prisma/client useless in this case?

Thanks

**Comment by janpio** [comment]
Yes, you use the explicit location of the package with `output` - `@prisma/client` is a convenience package that imports from the default `node_modules/.prisma/client` output location.

> It works but it creates a node_module & 2 package*.json in the db folder and not in dashboard/site

I think that should not be the case. You can go anywhere relative to the schema file in `output` and it should create that folder. If that is not the case, please open a bug report issues so we can discuss this with the details necessary. Thanks.

**Comment by birgersp** [comment]
I am now using a slightly different workaround, using an env variable that I set before I invoke `prisma generate`:

```
generator client {
    provider = "prisma-client-js"
	output = env("PRISMA_OUT")
}
```

That way, I can do

```
export PRISMA_OUT=../../myproject/node_modules/@prisma/client
npx prisma generate
```

to generate the client for my `myproject` directory.

However I don't think this is perfect, because the "shared" folder (under which the `schema.prisma` file resides) will then get its own set of packages in `node_modules`. And they are never used unless I do another `prisma generate`.

Personally I find it somewhatt un-intuitive that `prisma generate` is using only the schema file location to determine where the client should be built. Would it not be more "normal" to use the current directory? Other cli tools I've ever encountered (e.g npm, protoc, gradle, tar) emit files at the current directory. So in comparison I wanted the client to be built at `/home/birger/someweirddir` I would either have to do

```
cd /home/birger/someweirddir
prisma generate --schema /home/birger/wherever-the-schema-is/schema.prisma
```

or use a flag (if the tool has one).

Anyways, the workaround is fine but I think others might also find that they want to manually override where `prisma generate` emits its files.

**Comment by antoineanorra** [comment]
I think we have the same problem? Look here : #13164

**Comment by fellnerse** [comment]
Is there any update on this?
